javalib/src/main/scala/java/util/concurrent/Future.scala

```diff
@@ -23,4 +24,100 @@ trait Future[V] {
   @throws[TimeoutException]
   def get(timeout: Long, unit: TimeUnit): V
 
+  // since JDK 19
+  def resultNow(): V = {
+    if (!isDone())
+      throw new IllegalStateException("Task has not completed")
+    var interrupted = false
+    try
+      while (true) {
+        try get()
+        catch {
+          case _: InterruptedException => interrupted = true
+          case _: ExecutionException =>
+            throw new IllegalStateException("Task completed with exception")
+          case _: CancellationException =>
+            throw new IllegalStateException("Task was cancelled")
+        }
+      }
+    finally {
+      if (interrupted)
+        Thread.currentThread().interrupt()
+    }
+    ??? // unreachable
+  }
+
+  // Since JDK 19
+  def exceptionNow(): Throwable = {
+    if (!isDone())
+      throw new IllegalStateException("Task has not completed")
+    if (isCancelled())
+      throw new IllegalStateException("Task was cancelled")
+    var interrupted = false
+    try
+      while (true) {
+        try {
+          get()
+          throw new IllegalStateException("Task completed with a result")
+        } catch {
+          case _: InterruptedException => interrupted = true
+          case e: ExecutionException   => e.getCause()
```

Comment:
it seems this never returns?
```suggestion
          case e: ExecutionException   => return e.getCause()
```

Reply:
Fixed and ensured by adding tests